tokio-trace/src/lib.rs

```diff
@@ -275,28 +275,28 @@ impl Hash for SpanInner {
 }
 
 impl<'a> IntoIterator for &'a Span {
-    type Item = (&'static str, &'a dyn Value);
-    type IntoIter = Box<Iterator<Item = (&'static str, &'a dyn Value)> + 'a>; // TODO: unbox
+    type Item = (&'a str, &'a dyn Value);
+    type IntoIter = Box<Iterator<Item = (&'a str, &'a dyn Value)> + 'a>; // TODO: unbox
     fn into_iter(self) -> Self::IntoIter {
         Box::new(self.fields())
     }
 }
 
-impl<'a> IntoIterator for &'a Event<'a> {
-    type Item = (&'static str, &'a dyn Value);
-    type IntoIter = Box<Iterator<Item = (&'static str, &'a dyn Value)> + 'a>; // TODO: unbox
+impl<'b, 'a: 'b> IntoIterator for &'a Event<'a, 'b> {
+    type Item = (&'a str, &'a dyn Value);
+    type IntoIter = Box<Iterator<Item = (&'a str, &'a dyn Value)> + 'a>; // TODO: unbox
     fn into_iter(self) -> Self::IntoIter {
         Box::new(self.fields())
     }
 }
 
 pub struct DebugFields<'a, I: 'a>(&'a I)
 where
-    &'a I: IntoIterator<Item = (&'static str, &'a dyn Value)>;
+    &'a I: IntoIterator<Item = (&'a str, &'a dyn Value)>;
 
 impl<'a, I: 'a> fmt::Debug for DebugFields<'a, I>
 where
-    &'a I: IntoIterator<Item = (&'static str, &'a dyn Value)>,
```

Comment:
Nice!

Reply:
Yeah. This is a consequence of the `'meta` lifetime parameter on `Event`s.